pandas/tseries/index.py

```diff
@@ -740,20 +740,30 @@ def __setstate__(self, state):
             raise Exception("invalid pickle state")
     _unpickle_compat = __setstate__
 
+    def _add_datelike(self, other):
+        # adding a timedeltaindex to a datetimelike
+        if other is tslib.NaT:
+            result = np.zeros(len(self), dtype=np.int64)
```

Comment:
not sure if their is a way to make this more generic (as you had to repeat very similar code a few times)